### ISSUE TITLE
Federation transmission loop swallows exceptions

### ISSUE THREAD

**Issue by richvdh** [matrix-org/synapse#4239]
It's impossible to debug what's going on here. It's lucky the admin has DEBUG enabled, otherwise we'd have even fewer clues:

```
2018-11-28 17:35:46,310 - synapse.storage.txn - 234 - DEBUG - federation_transaction_transmission_loop-8- [TXN START] {get_new_device_msgs_for_remote-161}
2018-11-28 17:35:46,339 - synapse.storage.txn - 290 - DEBUG - federation_transaction_transmission_loop-8- [TXN END] {get_new_device_msgs_for_remote-161} 0.028580 sec
2018-11-28 17:35:46,373 - synapse.storage.txn - 234 - DEBUG - federation_transaction_transmission_loop-8- [TXN START] {get_devices_by_remote-163}
2018-11-28 17:35:46,555 - synapse.storage._base - 282 - DEBUG - federation_transaction_transmission_loop-8- [TXN FAIL] {get_devices_by_remote-163} (u'@<redacted>', u'FLXXHDQTMR')
2018-11-28 17:35:46,556 - synapse.storage.txn - 290 - DEBUG - federation_transaction_transmission_loop-8- [TXN END] {get_devices_by_remote-163} 0.183082 sec
2018-11-28 17:35:46,569 - synapse.federation.transaction_queue - 525 - WARNING - federation_transaction_transmission_loop-8- TX [matrix.org] Failed to send transaction: (u'<redacted>', u'FLXXHDQTMR')
```

(I suspect the root cause might be related to #4238, but there is no way to confirm this)

**Comment by andreaspeters** [comment]
Thanks @richvdh that u open this issue for me. :-) Well, this problem means, with our HS is it not possible to communicate with other HS in the federation and thats already since v0.33.2. So, its actually brakes our HS.

**Comment by andreaspeters** [comment]
something new about this issues? :-D

**Comment by andreaspeters** [comment]
Hi @richvdh I send u new logs. I hope the messages are more helpfull now.

**Comment by richvdh** [comment]
thanks @andreaspeters. I've opened https://github.com/matrix-org/synapse/issues/4419 to track your problem.

This bug (that the logs were useless) has been resolved by #4358.
